malcolm/modules/scanning/controllers/runnablecontroller.py

```diff
@@ -466,7 +470,15 @@ def do_run(self, hook):
         self.abortable_transition(ss.POSTRUN)
         completed_steps = self.configured_steps.value
         if completed_steps < self.total_steps.value:
-            steps_to_do = self.steps_per_run
+            if len(self.steps_per_run) > 1:
+                if self._breakpoint < len(self.steps_per_run):
+                    steps_to_do = self.steps_per_run[self._breakpoint]
+                    self._breakpoint += 1
```

Comment:
You increment it here, but you never set it back to zero (or one)

Reply:
I added an initialization line for the index in on_configure() function.
Additionally, it can be also rewound in pause().

Reply:
The index does get modified as Dino's comment above.